### DISCUSSION TITLE
Jacobian question on heat_conduction module

### DISCUSSION THREAD

**Discussion by jsevic** [idaholab/moose#28648] [Q&A Modules: General]
Hello,

I need a Joule heating kernel that allows coupling of instantaneous electrical conductivity so I made a new kernel that allows
sigma to be a coupled variable.

My kernel computation is identical the JouleHeatingSource kernel .C file however my Jacobian term is missing this term _delec_cond_dT[_qp] and as best as I can tell it looks like this is being computed based on constants from material properties.

Thank you!

John Sevic

**Comment by GiudGiud** [comment]
Hello

Are you saying because the material properties are constants you expect this term to be 0?
Does this term come into the definition of the jacobian?

Guillaume

**Reply by jsevic** [reply]
Hello,

Yes the material properties in this case would be constant, but the electrical conductivity is not constant it's a coupled variable.

Thank you.

For my kernel the Jacobian is 

`return -1*_D[_qp]*_grad_V[_qp]*_grad_V[_qp]*_phi[_j][_qp]*_test[_i][_qp]; `

and for the built-in Joule heating kernel, JouleHeatingSource, that does not allow for a coupled electrical conductivity, but does depend on material properties specified by mateirlaProperties, the Jacobian is

`  return -_delec_cond_dT[_qp] * _grad_elec[_qp] * _grad_elec[_qp] * _phi[_j][_qp] * _test[_i][_qp];
`

**Reply by GiudGiud** [reply]
are you using AD?
The JouleHeatingSource also has off-diagonal terms

**Reply by jsevic** [reply]
We have not tried AD, would you suggest that for a problem like this?

Thx!

**Reply by GiudGiud** [reply]
how many variables do you have?

you made this kernel

> a new kernel that allows sigma to be a coupled variable.

did you add the off-diagonal term for this?
AD can let you do this easily, but it's a little slower and it requires transitioning all the material properties to AD type

**Comment by jsevic** [comment]
OK I have found out the issue I had thank you for your help!
John Sevic
